aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -541,6 +541,23 @@ static void checkBasicAsStridedValidForSlice(
       "rewrite the `as_strided` call as a sequence of PyTorch view operations");
 }
 
+Tensor _new_zeros_with_same_feature_meta_batching_rule(
+    const Tensor& self,
+    const Tensor& other,
+    int64_t unused_num_batch_dims) {
+  TORCH_CHECK(isBatchedTensor(self) && !isBatchedTensor(other),
+    "Only the 'batched grad' use case is supported in PyTorch core.");
+
+  TORCH_INTERNAL_ASSERT(unused_num_batch_dims == 0,
+    "num_batch_dims should not be explicitly passed in because it will be overridden");
```

Comment:
When we get this to work with functorch we'll need to remove the assert, because if we have a BatchedTensor(BatchedTensor(x)) and call _new_zeros_with_same_feature_meta:
- the batching rule will call _new_zeros_with_same_feature_meta(BatchedTensor(x), other, 1)
- this will trigger the assert

But this is fine for now.